doc/book/zend.authentication.intro.md

```diff
@@ -0,0 +1,427 @@
+developer wishes to distinguish among several authentication result types. This allows developers to
+maintain detailed authentication result statistics, for example. Another use of this feature is to
+provide specific, customized messages to users for usability reasons, though developers are
+encouraged to consider the risks of providing such detailed reasons to users, instead of a general
+authentication failure message. For more information, see the notes below.
+- `getIdentity()`- returns the identity of the authentication attempt
+- `getMessages()`- returns an array of messages regarding a failed authentication attempt
+
+A developer may wish to branch based on the type of authentication result in order to perform more
+specific operations. Some operations developers might find useful are locking accounts after too
+many unsuccessful password attempts, flagging an IP address after too many nonexistent identities
+are attempted, and providing specific, customized authentication result messages to the user. The
+following result codes are available:
+
+```php
+<?php
+use Zend\Authentication\Result;
+
+Result::SUCCESS
+Result::FAILURE
+Result::FAILURE_IDENTITY_NOT_FOUND
+Result::FAILURE_IDENTITY_AMBIGUOUS
+Result::FAILURE_CREDENTIAL_INVALID
+Result::FAILURE_UNCATEGORIZED
+```
+
+The following example illustrates how a developer may branch on the result code:
+
+```php
+<?php
+// inside of AuthController / loginAction
+$result = $this->auth->authenticate($adapter);
+
+switch ($result->getCode()) {
+
+    case Result::FAILURE_IDENTITY_NOT_FOUND:
+        /** do stuff for nonexistent identity **/
+        break;
+
+    case Result::FAILURE_CREDENTIAL_INVALID:
+        /** do stuff for invalid credential **/
+        break;
+
+    case Result::SUCCESS:
+        /** do stuff for successful authentication **/
+        break;
+
+    default:
+        /** do stuff for other failure **/
+        break;
+}
+
+```
+
+## Identity Persistence
+
+Authenticating a request that includes authentication credentials is useful per se, but it is also
+important to support maintaining the authenticated identity without having to present the
+authentication credentials with each request.
+
+*HTTP* is a stateless protocol, however, and techniques such as cookies and sessions have been
+developed in order to facilitate maintaining state across multiple requests in server-side web
+applications.
+
+### Default Persistence in the PHP Session
+
+By default, `Zend\Authentication` provides persistent storage of the identity from a successful
+authentication attempt using the *PHP* session. Upon a successful authentication attempt,
+`Zend\Authentication\AuthenticationService::authenticate()` stores the identity from the
+authentication result into persistent storage. Unless specified otherwise,
+`Zend\Authentication\AuthenticationService` uses a storage class named
+`Zend\Authentication\Storage\Session`, which, in turn, uses [Zend\\Session ](zend.session). A custom
```

Comment:
Crap. Looks like we need to adjust the PCRE to:

```php
$docs = preg_replace('/:ref:`(.*?) <([^>]+)>`/', '[$1]($2)', $docs);
```

(links have the sequence ` <`; previous regex wasn't taking into account the whitespace before the `<`, which leads to extra whitespace inside the link text).